docs/sdk2/usage/PRODUCT_SYNC.md

```diff
@@ -0,0 +1,471 @@
+
+The `customObject` will have a `container:` **`"commercetools-sync-java.UnresolvedReferencesService.productDrafts"`**
+and a `key` representing the key of the productDraft that is waiting to be created/updated.
+
+
+Here is an example of a `CustomObject` in the target project that represents a productDraft with `productKey1`.  
+It being persisted as `CustomObject` means that the referenced productDrafts with keys `foo` and `bar` do not exist yet.
+
+```json
+{
+  "id": "d0fbb69e-76e7-4ec0-893e-3aaab6f4f6b6",
+  "version": 1,
+  "container": "commercetools-sync-java.UnresolvedReferencesService.productDrafts",
+  "key": "productKey1",
+  "value": {
+    "dependantProductKeys": [
+      "foo",
+      "bar"
+    ],
+    "productDraft": {
+      "productType": {
+        "typeId": "product-type",
+        "id": "main-product-type"
+      },
+      "masterVariant": {
+          "id": 1,
+          "sku": "white-shirt-1",
+          "key": "white-shirt-1",
+          "prices": [],
+          "images": [],
+          "attributes": [
+            {
+              "name": "product-reference-set",
+              "value": [
+                {
+                  "typeId": "product",
+                  "id": "foo"
+                },
+                {
+                  "typeId": "product",
+                  "id": "bar"
+                }
+              ]
+            }
+          ]
+        },
+      "key": "productKey1"
+    }
+  },
+  "createdAt": "2019-09-27T13:45:35.495Z",
+  "lastModifiedAt": "2019-09-27T13:45:35.495Z",
+  "lastModifiedBy": {
+    "clientId": "8bV3XSW-taCpi873-GQTa8lf",
+    "isPlatformClient": false
+  },
+  "createdBy": {
+    "clientId": "8bV3XSW-taCpi873-GQTa8lf",
+    "isPlatformClient": false
+  }
+}
+```
+
+As soon, as the referenced productDrafts are supplied to the sync, the draft will be created/updated and the 
+`CustomObject` will be removed from the target project.
+
+Keeping the old custom objects around forever can negatively influence the performance of your project and the time it takes to restore it from a backup.  Deleting unused data ensures the best performance for your project. Please have a look into the [Cleanup guide](CLEANUP_GUIDE.md) to cleanup old unresolved custom objects.
+
+##### More examples of how to use the sync
+
+1. [Sync from another commercetools project as a source](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/ctpprojectsource/products/ProductSyncIT.java).
+2. [Sync from an external source](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncIT.java). 
+3. [Sync with blacklisting/whitelisting](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/externalsource/products/ProductSyncFilterIT.java).
```

Comment:
Do you want to point the tests to the sdk v2 tests as well?

Reply:
I haven't done it yet, because i think probably the paths will change anyways. I will add a TODO for later adjustment.